sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c

```diff
@@ -230,11 +218,17 @@ static void _send(ng_pktsnip_t *pkt)
             return;
         }
 
-        sixlowpan->next = ipv6;
+        sixlowpan->next = pkt->next;
         pkt->next = sixlowpan;
         disp = sixlowpan->data;
         disp[0] = NG_SIXLOWPAN_UNCOMPRESSED;
-        payload_len++;
+        /* payload length and datagram size are different in that the payload
+         * length is the length of the IPv6 datagram + 6LoWPAN dispatches,
+         * while the datagram size is the size of only the IPv6 datagram */
+        payload_len = ng_pkt_len(pkt->next);
+        /* cppcheck: datagram_size will be read by ng_sixlowpan_frag implementation */
+        /* cppcheck-suppress unreadVariable */
+        datagram_size = (uint16_t)payload_len - 1;
```

Comment:
Can't we move this duplicated below the if-else blocks? (Setting a temp variable for the `-1`.)